### PR TITLE
Alphabet --- First attempt

### PR DESCRIPTION
**Pull request by bprzybylski**
This is the first attempt on the alphabet implementation.

### CODE REVIEW

File: src/Alphabet.jl [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+Alphabet of String:
+    1. "a" = ("c")⁻¹
+    2. "b"
+    3. "c" = ("a")⁻¹
+
+julia> get_symbol(A, -get_index(A, "a"))
+"c"
+```
+"""
+function get_symbol(A::Alphabet{T}, p::Int) where T
```
Comment (kalmarek) [inline src/Alphabet.jl:154]:
```suggestion
function Base.get_index(A::Alphabet{T}, p::Integer) where T
```
slightly more general signature and easier access, through standard brackets

File: src/Alphabet.jl [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    if p > 0
+        return A.alphabet[p]
+    elseif p < 0 && A.inversions[-p] > 0
+        return A.alphabet[A.inversions[-p]]
+    end
+    nothing
```
Comment (kalmarek) [inline src/Alphabet.jl:160]:
```suggestion
   throw(DomainError("Inverse to $(A.alphabet[-p]) is not defined"))
```

File: src/Alphabet.jl (hunk 1 above)
Comment (bprzybylski) [reply src/Alphabet.jl:154]: Yes, but the function does not return an index, so why to change the name to `get_index`?

Comment (kalmarek) [reply src/Alphabet.jl:154]: because then we can write `A[1]` for the first symbol and `A[-2]` for the inverse of the second

Comment (kalmarek) [reply src/Alphabet.jl:154]: oh, and of course it's `Base.getindex` :P

Comment (bprzybylski) [reply src/Alphabet.jl:154]: That is so clever!

**Pull request merged**